ansible/roles/test/tasks/pfc_wd/functional_test/functional_test.yml

```diff
@@ -66,11 +68,217 @@
+
+    - name: "Check ingress drop, rx port {{pfc_wd_test_port}}"
+      include: roles/test/tasks/ptf_runner.yml
+      vars:
+        ptf_test_name: PFC WD test
+        ptf_test_dir: ptftests
+        ptf_test_path: pfc_wd.PfcWdTest
+        ptf_platform: remote
+        ptf_test_params:
+          - testbed_type='{{testbed_type}}'
+          - router_mac='{{ansible_Ethernet0['macaddress']}}'
+          - queue_index='{{pfc_queue_index}}'
+          - pkt_count='{{pfc_wd_test_pkt_count}}'
+          - port_src='{{pfc_wd_test_port_id}}'
+          - ip_src='{{pfc_wd_test_neighbor_addr}}'
+          - ip_dst='{{pfc_wd_rx_neighbor_addr}}'
+          - wd_action='drop'
+        ptf_extra_options: "--relax --debug info --log-file /tmp/pfc_wd.PfcWdTest.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}.log "
+
+    - name: "Send 10K packets via {{pfc_wd_test_port}} to verify that other queue is not affected"
+      include: roles/test/tasks/ptf_runner.yml
+      vars:
+        ptf_test_name: PFC WD test
+        ptf_test_dir: ptftests
+        ptf_test_path: pfc_wd.PfcWdTest
+        ptf_platform: remote
+        ptf_test_params:
+          - testbed_type='{{testbed_type}}'
+          - router_mac='{{ansible_Ethernet0['macaddress']}}'
+          - queue_index='{{pfc_queue_index - 1}}'
+          - pkt_count='{{pfc_wd_test_pkt_count}}'
+          - port_src='{{pfc_wd_rx_port_id}}'
+          - ip_src='{{pfc_wd_rx_neighbor_addr}}'
+          - ip_dst='{{pfc_wd_test_neighbor_addr}}'
+          - wd_action='forward'
+        ptf_extra_options: "--relax --debug info --log-file /tmp/pfc_wd.PfcWdTest.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}.log "
+
+    - name: "Send 10K packets to {{pfc_wd_test_port}} to verify that other pg is not affected"
+      include: roles/test/tasks/ptf_runner.yml
+      vars:
+        ptf_test_name: PFC WD test
+        ptf_test_dir: ptftests
+        ptf_test_path: pfc_wd.PfcWdTest
+        ptf_platform: remote
+        ptf_test_params:
+          - testbed_type='{{testbed_type}}'
+          - router_mac='{{ansible_Ethernet0['macaddress']}}'
+          - queue_index='{{pfc_queue_index - 1}}'
+          - pkt_count='{{pfc_wd_test_pkt_count}}'
+          - port_src='{{pfc_wd_test_port_id}}'
+          - ip_src='{{pfc_wd_test_neighbor_addr}}'
+          - ip_dst='{{pfc_wd_rx_neighbor_addr}}'
+          - wd_action='forward'
+        ptf_extra_options: "--relax --debug info --log-file /tmp/pfc_wd.PfcWdTest.{{lookup('pipe','date +%Y-%m-%d-%H:%M:%S')}}.log "
+
+    - set_fact:
+        test_expect_file: "expect_pfc_wd_restore"
+
+    - name: Initialize loganalyzer
+      include: roles/test/files/tools/loganalyzer/loganalyzer_init.yml
+
+    - name: Stop PFC storm on fanout switch
+      action: apswitch template="{{pfc_wd_storm_stop_template}}"
+      args:
+        host: "{{peer_mgmt}}"
+        login: "{{peer_login}}"
+      connection: switch
+      when: peer_hwsku | search("Arista") or peer_hwsku | search("arista")
```

Comment:
>when: peer_hwsku | search("Arista") or peer_hwsku | search("arista") [](start = 6, length = 68)

stop pfc storm should be called for all fanout platforms, not just arista, right?

Reply:
I agree. Currently mlnx platform doesn't have stop pfc storm template, so I add this condition to prevent breaking their test env.  I will remove this condition to enforce all the platform follow the same logic.

Reply:
can you add an empty one for them so that it does not break their platform?